stats/view/benchmark_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package view
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+)
+
+var (
+	m, _  = stats.Float64("m", "", "")
+	k1, _ = tag.NewKey("k1")
+	k2, _ = tag.NewKey("k2")
+	k3, _ = tag.NewKey("k3")
+	k4, _ = tag.NewKey("k4")
+	k5, _ = tag.NewKey("k5")
+	k6, _ = tag.NewKey("k6")
+	k7, _ = tag.NewKey("k7")
+	k8, _ = tag.NewKey("k8")
+	view  = &View{
+		Measure:     m,
+		Aggregation: DistributionAggregation{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+		TagKeys:     []tag.Key{k1, k2},
+	}
+)
+
+// BenchmarkRecordReqCommand benchmarks calling the internal recording machinery
+// directly.
+func BenchmarkRecordReqCommand(b *testing.B) {
+	var (
+		w         = newWorker()
+		errchan   = make(chan error, 1)
+		subscribe = &subscribeToViewReq{views: []*View{view}, err: errchan}
+	)
+	subscribe.handleCommand(w)
+	err := <-errchan
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	var (
+		ctx  = context.Background()
+		ctxs = make([]context.Context, 0, 10)
```

Comment:
Either convert size into a const.

```
const size = 10
```

Or initiate this slice without a capacity. We don't care about the performance of this section of the test.

It makes it harder to refactor to hardcode size in multiple places.